dataclients/kubernetes/definitions/definitions.go

```diff
@@ -0,0 +1,103 @@
+/* Package definitions provides type definitions, parsing, marshaling and
+validation for Kubernetes resources used by Skipper. */
+package definitions
+
+import (
+	"encoding/json"
+
+	"github.com/go-yaml/yaml"
+	"github.com/pkg/errors"
+)
+
+type RouteGroupList struct {
+	Items []*RouteGroupItem `json:"items"`
+}
+
+type RouteGroupItem struct {
+	Metadata *Metadata       `json:"metadata"`
+	Spec     *RouteGroupSpec `json:"spec"`
+}
+
+type IngressList struct {
+	Items []*IngressItem `json:"items"`
+}
+
+type IngressItem struct {
+	Metadata *Metadata    `json:"metadata"`
+	Spec     *IngressSpec `json:"spec"`
+}
+
+// ParseRouteGroupsJSON parses a json list of RouteGroups into RouteGroupList
+func ParseRouteGroupsJSON(d []byte) (RouteGroupList, error) {
+	var rl RouteGroupList
+	err := json.Unmarshal(d, &rl)
+	return rl, err
+}
+
+// ParseRouteGroupsYAML parses a YAML list of RouteGroups into RouteGroupList
+func ParseRouteGroupsYAML(d []byte) (RouteGroupList, error) {
+	var rl RouteGroupList
+	err := yaml.Unmarshal(d, &rl)
+	return rl, err
+}
+
+// ValidateRouteGroup validates a RouteGroupItem
+func ValidateRouteGroup(rg *RouteGroupItem) error {
+	return rg.validate()
+}
+
+// ValidateRouteGroups validates a RouteGroupList
+func ValidateRouteGroups(rl *RouteGroupList) error {
+	var err error
+	// avoid the user having to repeatedly validate to discover all errors
+	for _, i := range rl.Items {
+		nerr := ValidateRouteGroup(i)
+		if nerr != nil {
+			err = errors.Wrap(err, nerr.Error())
```

Comment:
Does this error contain `namespace` and `name` of the resource on error?
This is a general question not need to be fixed in this PR, but we should make errors clear which resource is reporting errors otherwise we and our users have a hard time to investigate.
Right now as far as I had to check errors we have very limited information to track back the resource. Basically we just tell row+col of the error, which no one can track back to a resource.

Reply:
I'm not sure actually. Marked the todos

Reply:
Great thanks!